<launch>

  <!-- ==================== -->
  <!-- GazelleSim Arguments -->
  <!-- ==================== -->
  <arg name="show_dynamic_reconfigure" default="true" />


  <!-- ========================= -->
  <!-- Set simulation parameters -->
  <!-- ========================= -->
  <param name="map_dir" value="$(find blob_follow)/map" />
  <rosparam command="load"
            file="$(find blob_follow)/map/reu24_performance.yaml" />

  <!-- ============= -->
  <!-- Define robots -->
  <!-- ============= -->
  <rosparam>
    robot_list: ["robot1"]
  </rosparam>
  
  <!-- Define robot1 -->
  <group ns="robot1">
    <rosparam command="load"
              file="$(find blob_follow)/robot/ACTor.yaml" />
    <rosparam>
        Xr_init:  0.0
        Yr_init:  0.0
        Theta_init: deg(0.0)
    </rosparam>
    <node name="blob" pkg="blob_follow" type="blob.py" output="screen">
        <param name="image_topic" value="camera/image_raw" />
        <param name="twist_topic" value="cmd_vel" />
    </node>
  </group>

  <node pkg="rqt_gui" name="rqt_gui" type="rqt_gui" />


  <!-- ================= -->
  <!-- Launch GazelleSim -->
  <!-- ================= -->
  <include file="$(find gazelle_sim)/launch/gazelle_sim.launch">
    <arg name="show_dynamic_reconfigure" default="true" />
  </include>


</launch>